### PR TITLE
Add crawler for codeql-plugin

### PR DESCRIPTION
**Pull request by issc29**
Crawler for downloading CodeQL version releases built by https://github.com/github/codeql-action/releases

Used by plugin: https://github.com/jenkinsci/codeql-plugin

### CODE REVIEW

**Comment by issc29** [comment]
can I get a quick 👍  I think this is the last PR I need before I release my plugin 😄

File: codeql.groovy [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import net.sf.json.*
+
+def wc = new WebClient();
+wc.addRequestHeader("accept", "application/json")
+wc.getOptions().setThrowExceptionOnFailingStatusCode(false);
+wc.getOptions().setPrintContentOnFailingStatusCode(false);
+
+def API_URL = "https://api.github.com";
+Page p = wc.getPage(API_URL + "/repos/github/codeql-action/releases")
```
Comment (daniel-beck) [inline codeql.groovy:15]: This looks brittle without API token being used

Comment (issc29) [reply codeql.groovy:15]: @daniel-beck a couple of [other crawlers](https://github.com/jenkins-infra/crawler/search?q=api) seem to do the same thing but happy to change it if it makes sense! Are there any GitHub API keys that get injected to the build by chance?

Comment (daniel-beck) [reply codeql.groovy:15]: Oh wow, I didn't know that we're already doing that.

Which is kind of even worse with the (very low IIRC) anonymous API limits of GH, the more crawlers use them, the more likely it is we hit the limit.

Not sure about a credential. Maybe something to do later if it doesn't work anymore after all.

Comment (timja) [reply codeql.groovy:15]: it may not even be rate limited. releases in general is CDN backed

Comment (issc29) [reply codeql.groovy:15]: if thats the case, can I get a 👍 😉

**Review by timja — APPROVED** [review]

**Pull request merged**